### ISSUE TITLE
Style/WhileUntilDo autocorrection creates a syntax error on 2 line while loop

### ISSUE THREAD

**Issue by Zopolis4** [rubocop/rubocop#15197]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/WhileUntilDo, Style/InfiniteLoop

### Steps to reproduce the problem

Create a `.rubocop.yml` file with the provided configuration.

Create `example.rb` with the following content:
```ruby
# frozen_string_literal: true

i = 0
while i < 3 do i += 1
end

while true
  break 123
end
```

Run `rubocop -A example.rb`

### Expected behavior

I expected it to not introduce a syntax error into working code.

### Actual behavior

```
$ cat example.rb
# frozen_string_literal: true

i = 0
while i < 3 do i += 1
end

while true
  break 123
end
$ ruby example.rb
$ rubocop -A example.rb
Inspecting 1 file
F

Offenses:

example.rb:4:13: F: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
while i < 3 i += 1
            ^
example.rb:4:13: C: [Corrected] Style/WhileUntilDo: Do not use do with multi-line while.
while i < 3 do i += 1
            ^^
example.rb:5:1: F: Lint/Syntax: unexpected token kEND
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
end
^^^
example.rb:7:1: C: [Corrected] Style/InfiniteLoop: Use Kernel#loop for infinite loops.
while true
^^^^^

1 file inspected, 4 offenses detected, 2 offenses corrected
$ ruby example.rb
example.rb: --> example.rb

expected a predicate expression for the `while` statement

> 3  i = 0
> 4  while i < 3 i += 1
> 5  end

example.rb:4: syntax error, unexpected local variable or method, expecting `do' for condition or ';' or '\n' (SyntaxError)
while i < 3 i += 1
            ^
$ cat example.rb
# frozen_string_literal: true

i = 0
while i < 3 i += 1
end

loop do
  break 123
end
```

### RuboCop configuration

```yaml
AllCops:
  DisabledByDefault: true
  TargetRubyVersion: 3.3
  
Style/WhileUntilDo:
  Enabled: true

Style/InfiniteLoop:
  Enabled: true
```

### RuboCop version

```shell
1.87.0 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 3.3, running on ruby 3.3.8) [x86_64-linux-gnu]
```
